### PR TITLE
feat(python): Add ArcTan2 to `SQLContext` 

### PR DESCRIPTION
**Pull request by SeanTroyUWO**
Missed this in #9453. Adds degree version too. 

Please let me know if there's anything else I should add. This could be exposed to the other entry points maybe.

### CODE REVIEW

**Comment by ritchie46** [comment]
Thanks, can you add it to the python API (expression and Series) and the reference guide as well?

**Comment by SeanTroyUWO** [comment]
> Thanks, can you add it to the python API (expression and Series) and the reference guide as well?

For sure. I actually just left on vacation, but if I'll be back to update it in around a week.

**Comment by SeanTroyUWO** [comment]
> Thanks, can you add it to the python API (expression and Series) and the reference guide as well?

Just got this done, thanks for waiting :).

File: py-polars/polars/functions/lazy.py [hunk 1]
```diff
@@ -1550,6 +1550,100 @@ def cumreduce(
     return wrap_expr(plr.cumreduce(function, exprs))
 
 
+def arctan2(y: str | Expr, x: str | Expr) -> Expr:
```
[the remark this answers is not in the document]
Comment (SeanTroyUWO) [reply py-polars/polars/functions/lazy.py:1553]: In that case, are we looking for an API that looks like:

`df_result_new? = df.select(pl.col("y").arctan2("x").alias("atan2"))`

As apposed to: 

`df_result_old = df.select(pl.arctan2("y","x").alias("atan2"))`

[the remark this answers is not in the document]
Comment (ritchie46) [reply py-polars/polars/functions/lazy.py:1553]: Oh, sorry my bad. I didn't realize there were 2 arguments. In that case this makes sense.

**Comment by ritchie46** [comment]
Thank you @SeanTroyUWO. Sorry for the confusion. This looks great! Can you do a rebase? Then it is good to go.

**Comment by SeanTroyUWO** [comment]
> Thank you @SeanTroyUWO. Sorry for the confusion. This looks great! Can you do a rebase? Then it is good to go.

Sure I can after the weekend. Thanks for the patience as always :).

**Comment by SeanTroyUWO** [comment]
> > Thank you @SeanTroyUWO. Sorry for the confusion. This looks great! Can you do a rebase? Then it is good to go.
> 
> Sure I can after the weekend. Thanks for the patience as always :).

**Comment by SeanTroyUWO** [comment]
@ritchie46 This should be good now. I accidentally closed this, so sorry about that.

**Comment by ritchie46** [comment]
Great!

**Pull request merged**